CHANGELOG.md

```diff
@@ -31,6 +31,7 @@ the `Marshaler<Writer>` transform overload that were _mistakenly_ passing
 `Writer` explicitly (e.g. `bond::Apply<Writer>(marshaler, value)`) will now
 get a compiler error. To fix, remove the `<Writer>` part:
 `bond::Apply(marshaler, value)`.
+* C++ codegen now generates [extern templates](http://en.cppreference.com/w/cpp/language/function_template) of `bond::Apply` instead of overloads.
```

Comment:
Even though we are removing function overloads from user namespace, which means a breaking change if they are calling with a fully qualified name (e.g. `MyNamespace::Apply(transform, value)`), I doubt anyone is doing that because it simply defeats the purpose of those functions. In fact, any call to `Apply` must be done with unqualified name in order the ADL to work. I wouldn't consider this as a breaking change, but let me know if you have a different opinion.